test/functional/helpers/createFixture/clientScripts.js

```diff
@@ -35,15 +36,22 @@ const localAlloyLibraryPath = path.join(
 const remoteAlloyLibraryUrl =
   "https://cdn1.adoberesources.net/alloy/latest/alloy.js";
 
-let localAlloyCode;
+// We use this getter for retrieving the library code instead of just loading
+// the library code a single time up-front, because we want every run to be
+// using the latest library code. This is important when developing in watch
+// mode and making changes to source files.
+const getLocalAlloyCode = () => {
+  // readCache caches file content until the file is modified, at which
+  // point it will retrieve fresh file content, cache it, and return it.
+  return readCache.sync(localAlloyLibraryPath, "utf8");
+};
 
-// We're only testing against the alloy code found in /dist when alloyEnv
-// is "int". Otherwise, we're testing against the prod alloy that's on the CDN
-// and in those cases alloy code may not even exist in /dist. This is why
-// we only try to load the file from the file system if alloyEnv is int.
-if (alloyEnv === "int") {
-  localAlloyCode = fs.readFileSync(localAlloyLibraryPath, "utf8");
```

Comment:
The changes in this file were to improve watch mode when running functional tests. I noticed that depending on what changes I was making in our source code and how I was focusing a particular test, that my changes weren't being used when the test was re-run. This was happening because we were reading the produced Alloy build file once using this line of code and then TestCafe wasn't running this line again when re-running the test in watch mode.

I got around this by making this a function (see `getLocalAlloyCode` in my changes) that reads the Alloy build file every time TestCafe injects scripts into the test page, which happens every time a test is run or re-run. I optimized it though by using the `read-cache` npm package, which caches a file's contents and returns that content as long as the file hasn't changed since the last call (it uses the file's last modified date to determine if it's changed rather than having to read all its contents again). That's just so we're not reading the full file contents from disk for every test.